javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -1013,44 +974,27 @@ public int length() {
     }
 
     @Override
-    public IndexedSeq<T> removeAll(Predicate<? super T> predicate) {
+    public Vector<T> removeAll(Predicate<? super T> predicate) {
```

Comment:
I removed that kind of tests (then re-added it after a few months and finally removed it again) - we have to take care of our own (or re-add the tests again).

The tests were too complicated and I had to maintain a white-list of invalid return types. This was error prone.